server/cache/cache_test.go

```diff
@@ -178,3 +178,79 @@ func (s *testRegionCacheSuite) TestFifoCache(c *C) {
 	cache.Remove()
 	c.Assert(cache.Len(), Equals, 0)
 }
+
+func (s *testRegionCacheSuite) TestARCCache(c *C) {
+	cache := NewARC(4)
+	cache.Put(1, "1")
+	cache.Put(2, "2")
+	cache.Put(3, "3")
+
+	c.Assert(cache.Len(), Equals, 3)
+
+	val, ok := cache.Get(3)
+	c.Assert(ok, IsTrue)
+	c.Assert(val, DeepEquals, "3")
+
+	val, ok = cache.Get(2)
+	c.Assert(ok, IsTrue)
+	c.Assert(val, DeepEquals, "2")
+
+	val, ok = cache.Get(1)
+	c.Assert(ok, IsTrue)
+	c.Assert(val, DeepEquals, "1")
+
+	cache.Put(4, "4")
+	cache.Put(4, "4")
+
+	c.Assert(cache.Len(), Equals, 3)
+
```

Comment:
better to add a comment like:
```
_ _ _ _ | _ _ _ _ | 4 1 2 _ | 3 _ _ _ |
    b1       t1        t2        b2
```

Reply:
I found some bug on ARC put, fix will be push later